contracts/SponsorFaucet.sol

```diff
@@ -49,12 +49,14 @@ contract SponsorFaucet is Ownable, Pausable, ReentrancyGuard {
         cpc.setSponsorForGas.value(gas_bound)(dapp, upper_bound);
         dapps[dapp].gas_cnt.add(1);
         dapps[dapp].limit.add(gas_bound);
+        emit applied(msg.sender, dapp, upper_bound);
```

Comment:
Why not record both `gas_bount` and `upper_bound`?

Moreover, how about `collateral_bound`?

Reply:
**applied** is used to track single applied fund.